crypto/merklearray/vectorCommitmentArray.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package merklearray
+
+import (
+	"fmt"
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/protocol"
+	"math/bits"
+)
+
+// ErrGetOutOfBound returned when trying to retrieve an element which is out of
+// the padded array bound.
+var (
+	ErrGetOutOfBound = "vectorCommitmentArray.Get(%d): out of bounds, full size %d"
```

Comment:
instead of defining an error string here, we should have a custom error object here, which has two fields : `requestIndex`, `length`. The Error() method should perform the formatting.
The unit test could verify this error by using `errors.Is`.